tests/integration/test_optislang/test_optislang_integration.py

```diff
@@ -36,69 +40,65 @@ def test_simple_solve(load_mixing_elbow_param_case_dat):
-            solver_session.tui.solve.dual_time_iterate()
-        else:
-            solver_session.tui.solve.iterate()
-
-        convergence = solver_session.scheme_eval.string_eval(
-            "(rpgetvar 'solution/converged?)"
-        )
-
-        # solution output (test conditional statement)
-        if convergence == "#f":  # -> let user know
-            print("Failed to converge")
-        else:
-            print("Solution is converged")
-
-        assert convergence == "#t", "Solution failed to converge"
-
-        # Step 6: Read the data again from the case and data file
-        solver_session.tui.file.read_case_data(case_path)
-        input_parameters2 = solver_session.scheme_eval.eval(
-            (Symbol("list-input-parameters"),)
-        )
-        output_parameters2 = solver_session.scheme_eval.eval(
-            (Symbol("list-output-parameters"),)
-        )
-        input_parameters2 = input_parameters2["inlet2_temp"]
-        output_parameters2 = output_parameters2["outlet_temp-op"]
-        assert input_parameters[0] == input_parameters2[0]
-        assert output_parameters[0] == output_parameters2[0]
-        solver_session.exit()
+    input_parameters = {}
+    for p in reader.input_parameters():
+        input_parameters[p.name] = (p.value, p.numeric_value, p.units)
+    output_parameters = {}
+    for o in reader.output_parameters():
+        output_parameters[o.name] = (0, o.units)
+    solver_session.settings.file.read_case(file_name=case_path)
+
+    input_parameters = input_parameters["inlet2_temp"]
+    output_parameters = output_parameters["outlet_temp-op"]
+
+    # Step 4: Set a variation on these input parameters
+    # variations/designs are generated by optiSLang based on
+    # algorithm selected
+    inputs_table = InputParameters(solver_session)
+    inputs_table["inlet2_temp"] = 600
+    solver_session.settings.file.write_case(file_name="design_elbow_param.cas.h5")
+
+    # Step 5: Solve
+    solver_session.settings.solution.initialization.standard_initialize()
+
+    # check if solution is steady or transient
+    workflow = solver_session.rp_vars("rp-unsteady?")
+
+    # iterate workflow
+    if workflow:
+        solver_session.settings.solution.run_calculation.dual_time_iterate()
+    else:
+        solver_session.settings.solution.run_calculation.iterate()
+
+    convergence = solver_session.rp_vars("solution/converged?")
+
+    # solution output (test conditional statement)
+    if not convergence:  # -> let user know
+        print("Failed to converge")
+    else:
+        print("Solution is converged")
+
+    assert convergence == True, "Solution failed to converge"
+
+    # Step 6: Read the data again from the case and data file
+    solver_session.settings.file.read_case_data(file_name=case_path)
+
+    inputs_table = InputParameters(solver_session)
+    assert input_parameters[0] == inputs_table["inlet2_temp"]
+
+    outputs_table = OutputParameters(solver_session)
```

Comment:
fluent_output_table = self.session.parameters.output_parameters.list()
for key, entry in fluent_output_table.items():
    value = entry[0] #0: number and 1: unit(String)
    unit = entry[1] #0: number and 1: unit(String)

Reply:
@tejalprabhu01 Should be sorted now